tools/nl/embeddings/build_embeddings_v2_test.py

```diff
@@ -0,0 +1,133 @@
+  overrides = ["", "", "override3", ""]
+  curated = ["abc1;xyz1", "abc2;xyz2;abc2", "abc3;xyz3", "abc4;xyz4"]
+
+  return pd.DataFrame.from_dict({
+      "dcid": dcids,
+      "Name": names,
+      "Description": decriptions,
+      "Override_Alternatives": overrides,
+      "Curated_Alternatives": curated,
+  })
+
+
+class TestEndToEnd(unittest.TestCase):
+
+  def testFailure(self):
+
+    # Mock the get_sheets_data function to return test data.
+    # Mocking with an empty DataFrame should result in a KeyError exception for
+    # the expected column names not found.
+    be.get_sheets_data = mock.Mock(return_value=pd.DataFrame())
+
+    # Given that the get_sheets_data() function is mocked, the Context
+    # object does not need a valid `gs` and `bucket` field.
+    ctx = be.Context(gs=None,
+                     model=SentenceTransformer("all-MiniLM-L6-v2"),
+                     bucket="",
+                     tmp="/tmp")
+
+    # Smilarly, sheets_url and worksheet_name can be empty strings.
+    sheets_url = ""
+    worksheet_name = ""
+
+    # Filepaths all correspond to the testdata folder.
+    local_sheets_csv_filepath = "testdata/output/sheets_data.csv"
+    local_merged_filepath = "testdata/output/merged_data.csv"
+    other_alternatives_filepath = "testdata/input/other_alternatives.csv"
+    palm_alternatives_filepath = "testdata/input/palm_alternatives.csv"
+
+    with self.assertRaises(KeyError):
+      be.build(ctx, sheets_url, worksheet_name, local_sheets_csv_filepath,
+               local_merged_filepath,
+               [other_alternatives_filepath, palm_alternatives_filepath])
+
+  def testSuccess(self):
+
+    # Mock the get_sheets_data function to return test data.
+    be.get_sheets_data = mock.Mock(return_value=get_test_sv_data())
+
+    # Given that the get_sheets_data() function is mocked, the Context
+    # object does not need a valid `gs` and `bucket` field.
+    ctx = be.Context(gs=None,
+                     model=SentenceTransformer("all-MiniLM-L6-v2"),
+                     bucket="",
+                     tmp="/tmp")
+
+    # Smilarly, sheets_url and worksheet_name can be empty strings.
+    sheets_url = ""
+    worksheet_name = ""
+
+    # Filepaths all correspond to the testdata folder.
+    local_sheets_csv_filepath = "testdata/output/sheets_data.csv"
+    local_merged_filepath = "testdata/output/merged_data.csv"
+    other_alternatives_filepath = "testdata/input/other_alternatives.csv"
+    palm_alternatives_filepath = "testdata/input/palm_alternatives.csv"
+
+    embeddings_df = be.build(
+        ctx, sheets_url, worksheet_name, local_sheets_csv_filepath,
+        local_merged_filepath,
+        [other_alternatives_filepath, palm_alternatives_filepath])
+
+    # Verify that the output directory has the expected files.
+    self.assertTrue(os.path.exists(os.path.join(local_sheets_csv_filepath)))
+    self.assertTrue(os.path.exists(os.path.join(local_merged_filepath)))
```

Comment:
I'm slightly puzzled... wonder why not set local_merged_filepath and local_sheets_csv_filepath to tmp-dir (eg `tempfile.TemporaryDirectory()`), and compare the "actual" with "expected" from testdata/?